test/tx_funs.erl

```diff
@@ -37,6 +36,15 @@
 -define(assertToFunThrow(Expected, Expression),
         ?assertThrow(Expected, ?make_standalone_fun(Expression))).
 
+%% The compiler is smart enough to optimize away many instructions by
+%% inspecting types and values. `mask/1' confuses the compiler by sending
+%% and receiving the value.
+-define(mask(Value),
+        begin
+            self() ! Value,
+            receive __Msg -> __Msg end
```

Comment:
The compiler gives a warning about duplicated bindings when using the mask macro twice in a scope. I didn't think this would be an issue since `__Msg` is only in-scope for the `receive` block, but it turns out that the binding leaks out of scope:

```erl
1> self() ! 1.
1
2> receive Msg -> Msg end.
1
3> Msg.
1
```

Is there a way of introducing hygienic bindings in a macro in Erlang?